webapp/app/forms/steps/lotse/personal_data.py

```diff
@@ -523,4 +523,136 @@ def render(self):
+
+    @root_validator(skip_on_failure=True)
+    def person_b_has_disability(cls, values):
+        if values.get('person_b_has_disability') != 'yes':
+            raise ValidationError
+        return values
+    
+class StepPauschbetragPersonA(LotseFormSteuerlotseStep):
+    name = 'person_a_requests_pauschbetrag'
+    section_link = SectionLink('mandatory_data', StepFamilienstand.name, _l('form.lotse.mandatory_data.label'))
+
+    preconditions = [PersonAHasDisabilityPrecondition]
+    
+    class InputForm(SteuerlotseBaseForm):
+        person_a_requests_pauschbetrag = SelectField(
+            # TODO: This is because of key mapping - the text is within the react component
+            choices=[('yes', 'yes'),('no', '')],
+            render_kw={'data_label':  _l('form.lotse.request_pauschbetrag.data_label')},         
+            validators=[InputRequired(_l('validate.input-required'))])
+        
+    @classmethod
+    def get_label(cls, data=None):
+        return ngettext('form.lotse.person_a.request_pauschbetrag.label', 'form.lotse.person_a.request_pauschbetrag.label',
+                        num=get_number_of_users(data))        
+    
+    def _get_value_representation(self, value):
+        result = ''
+        
+        if value == 'yes':
+            result = self.get_pauschbetrag() + ' ' + _('currency.euro')
+            
+        return result
+    
+    def render(self):
+        props_dict = PauschbetragPersonProps(            
+            step_header={
+                'title': ngettext('form.lotse.person_a.request_pauschbetrag.title', 'form.lotse.person_a.request_pauschbetrag.title',
+                        num=get_number_of_users(self.stored_data))
+            },
+            form={
+                'action': self.render_info.submit_url,
+                'csrf_token': generate_csrf(),
+                'show_overview_button': bool(self.render_info.overview_url),
+            },
+            pauschbetrag=self.get_pauschbetrag(),
+            fields=form_fields_dict(self.render_info.form),
+            prev_url=self.render_info.prev_url
+        ).camelized_dict()
+        
+        
+        # Humps fails to camelize individual letters correctly, so we have to fix it manually.
+        # (A fix exists but hasn't been released at the time of writing: https://github.com/nficano/humps/issues/61)
+        props_dict['fields']['personARequestsPauschbetrag'] = props_dict['fields'].pop('personA_requestsPauschbetrag')
+        
+
+        return render_template('react_component.html',
+                               component='PauschbetragPersonAPage',
+                               props=props_dict,
+                               form=self.render_info.form,
+                               header_title=_('form.lotse.header-title'))
+        
+    def get_pauschbetrag(self):
+        # TODO: Replace with backend logic
+        return "1"
+        
+class StepPauschbetragPersonB(LotseFormSteuerlotseStep):
```

Comment:
Doesn't this share a lot with `StepPauschbetragPersonA`? Especially, `_get_value_representation` and `get_pauschbetrag`?

Reply:
Yes, but the precondition and some texts are different.

Reply:
Okay, but why not define them only once either through inheritance or as private functions?